cleanlab_studio/internal/clean_helpers.py

```diff
@@ -1,15 +1,17 @@
+import itertools
 import time
 from typing import Optional
+
 from tqdm import tqdm
 
+from cleanlab_studio.errors import CleansetError
 from cleanlab_studio.internal.api import api
 
 
-def poll_cleanset_status(api_key: str, cleanset_id: str, timeout: Optional[int] = None) -> bool:
+def poll_cleanset_status(api_key: str, cleanset_id: str, timeout: Optional[float] = None) -> None:
     start_time = time.time()
     res = api.get_cleanset_status(api_key, cleanset_id)
-    spinner = "|/-\\"
-    spinner_idx = 0
+    spinner = itertools.chain("|/-\\")
```

Comment:
Would be nice to factor out spinner into a reusable utility, rather than having special spinner code written or copy-pasted into these kinds of helper functions. But this could be done separately from this PR.

Reply:
I started doing that, but didn't want to spend the extra time. This is the only place we use the spinner -- can refactor out the next time